src/main/java/org/springframework/data/semantic/support/cache/EhCacheEntityCache.java

```diff
@@ -42,12 +42,12 @@ public EhCacheEntityCache(SemanticMappingContext mappingContext, CacheManager ca
 	@Override
 	public <T> void remove(T entity) {
 		Ehcache cache = getCache(entity.getClass());
-		cache.remove(getId(entity).toString());
+		cache.remove(getId(entity).stringValue());

```

Comment:
Here we must use toString instead as it was in the original code. toString and stringValue give different results for datatyped literals and probably other model objects.